main.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+)
+
+type project struct{}
+
+type Project interface {
+	ReadInput() (<-chan Command, []string)
+	Execute(cmds <-chan Command, input []string) []string
+	PrintResult(res []string)
+}
+
+var app Project
+
+type Command struct {
+	Name     string
+	Argument string
+}
+
+const (
+	Cat  = "cat"
+	Grep = "grep"
+	Sort = "sort"
+)
```

Comment:
Should have own type e.g.

```go
type TxnStatus string

const (
  Open         TxnStatus = "TXN_OPEN"
  Discarded TxnStatus = "TXN_DISCARDED"
```

In this way we are making sure that only pre-defined values could be used